packages/itwinui-react/src/core/Tabs/Tabs.tsx

```diff
@@ -526,85 +503,399 @@ export const Tabs = (props: TabsProps) => {
-          },
-          tabsClassName,
-        )}
-        data-iui-overflow={overflowOptions?.useOverflow}
-        data-iui-scroll-placement={scrollingPlacement}
-        role='tablist'
-        ref={refs}
-        onKeyDown={onKeyDown}
+        className={cx('iui-tabs-actions-wrapper', className)}
+        ref={ref}
         {...rest}
       >
-        {labels.map((label, index) => createTab(label, index))}
+        {children}
       </Box>
+    );
+  } else {
+    return <></>;
+  }
+}) as PolymorphicForwardRefComponent<'div'>;
+TabsActions.displayName = 'Tabs.Actions';
 
-      {actions && (
-        <Box className='iui-tabs-actions-wrapper'>
-          <Box className='iui-tabs-actions'>{actions}</Box>
-        </Box>
-      )}
+// ----------------------------------------------------------------------------
+// Tabs.Action component
 
-      {children && (
-        <Box
-          className={cx('iui-tabs-content', contentClassName)}
-          role='tabpanel'
-        >
-          {children}
-        </Box>
-      )}
-    </Box>
+const TabsAction = React.forwardRef((props, ref) => {
+  const { className, children, ...rest } = props;
+
+  const { type } = useSafeContext(TabsContext);
+
+  if (type !== 'pill') {
+    return (
+      <Box className={cx('iui-tabs-actions', className)} ref={ref} {...rest}>
+        {children}
+      </Box>
+    );
+  } else {
+    return <></>;
+  }
+}) as PolymorphicForwardRefComponent<'div'>;
+TabsAction.displayName = 'Tabs.Action';
+
+// ----------------------------------------------------------------------------
+// Tabs.Panels component
+
+const TabsPanels = ({
+  children,
+}: {
+  children: React.ReactNode[] | React.ReactNode;
+}) => {
+  const { currentActiveIndex } = useSafeContext(TabsContext);
+
+  const items = Array.isArray(children) ? children : [children];
+
+  return (
+    <>
+      {items.map((item, index) => {
+        return (
+          <TabsContext.Provider
+            value={{ index, currentActiveIndex }}
+            key={index}
+          >
+            {item}
+          </TabsContext.Provider>
+        );
+      })}
+    </>
   );
 };
+TabsPanels.displayName = 'Tabs.Panels';
+
+// ----------------------------------------------------------------------------
+// Tabs.Panel component
+
+const TabsPanel = React.forwardRef((props, ref) => {
+  const { className, children, ...rest } = props;
+
+  const { index, currentActiveIndex } = useSafeContext(TabsContext);
+
+  if (index === currentActiveIndex) {
+    return (
+      <Box
+        className={cx('iui-tabs-content', className)}
+        role='tabpanel'
+        ref={ref}
+        {...rest}
+      >
+        {children}
+      </Box>
+    );
+  } else {
+    return <></>;
+  }
+}) as PolymorphicForwardRefComponent<'div'>;
+TabsPanel.displayName = 'Tabs.Panel';
+
+/**
+ * Tabs organize and allow navigation between groups of content that are related and at the same level of hierarchy.
+ * @example
+ * <Tabs>
+ *   <Tabs.Tab>
```

Comment:
missing `<Tabs.TabList>` for this example to work

Reply:
Good catch! This has been fixed